app/components/bulk_upload_error_summary_table_component.rb

```diff
@@ -0,0 +1,17 @@
+class BulkUploadErrorSummaryTableComponent < ViewComponent::Base
+  attr_reader :bulk_upload
+
+  def initialize(bulk_upload:)
+    @bulk_upload = bulk_upload
+
+    super
+  end
+
+  def sorted_errors
+    @sorted_errors ||= bulk_upload
+      .bulk_upload_errors
+      .group(:col, :field, :error)
+      .count
+      .sort_by { |el| el[0][0].rjust(3, "0") }
```

Comment:
can this/these be a scope(s)?

Reply:
the `sort_by` can't be a scope as its ruby and SQL. I'm not aware of how to change or apply a different sort algorithm to postgres. I don't see the benefit of extracting the whole query to a scope as it's not used anywhere else

Reply:
or maybe it could be a scope but with my latter point feel it's kind of moot

Reply:
I guess it might be quicker if it's sql, but not sure how easy it would be to translate, so will trust you on that one